client/python/apache_polaris/cli/constants.py

```diff
@@ -258,6 +260,12 @@ class Create:
                 "(Only for S3) Indicates that Polaris should not use STS (e.g. if STS is not available)"
             )
             PATH_STYLE_ACCESS = "(Only for S3) Whether to use path-style-access for S3"
+            KMS_KEY_CURRENT = (
+                "(Only for AWS S3) The AWS KMS key ARN to be used for encrypting new S3 data"
```

Comment:
I am assuming this is required because we need to use this key to encrypt metadata.json ? as when we are vending creds we don't know which snapshot the client will be reading so we vend creds for all or we just give decrypt creds for allowed key and encrypt | decrypt creds for current keys ?

Reply:
Polaris does not use KMS keys directly. It only generates AWS policies that allow those keys to be used on the AWS side when S3 requests are made. But, yes, the current key is used for writing new data. Zero or more additional keys are also allowed to be used because they might be required for dealing with old files.

Reply:
> Polaris does not use KMS keys directly.

wouldn't we be needing this for encrypting / decrypting metadata.json ? 


> additional keys are also allowed to be used because they might be required for dealing with old files

I agree with additional keys but my question was why would Polaris vends creds for old kms keys for encrypting, files are immutable, so old keys should be vended for decrypt, similarly new key should have encrypt / decrypt. 
Do we vend creds for encryption and decryting for all key in our sts policy ?

Reply:
> my question was why would Polaris vends creds for old kms keys for encrypting, [...]

Currently it does. However, this is beyond the scope of current PR (CLI). It's about the actual java code from #2802 :)

Normally, I'd think "additional" keys should get only decryption rights, but this may be tricky from the manual key rotation perspective.